dslib.h

```diff
@@ -104,6 +105,30 @@ char* top(stack *top)
 		return top->d.p;
 }
 
+void emptystack(stack **top)
+{
+	if(*top == NULL)
+		return;
+		
+	while(*top != NULL) {
+		stack *dt = *top;
+		*top = (*top)->link;
+		free(dt);
+	}
+}
+
+void cleanqueue(queue **front, queue **rear)
+{
+	if (*front != NULL && *rear != NULL) {
```

Comment:
rear pointer is only useful in enqueue() because we don't want to traverse the full list to find the end. But that's about it. Update it but don't check it. front NULL implies there's nothing in the queue.

If there's no person at the front of a real queue can there be any person at the end? Is there any point in checking the condition each time?